lib/improver/nbhood/nbhood.py

```diff
@@ -170,7 +171,253 @@ def __repr__(self):
         return result.format(self.unweighted_mode)
 
     @staticmethod
-    def mean_over_neighbourhood(cube, cells_x, cells_y, nan_mask):
+    def cumulate_array(cube):
+        """
+        Method to calculate the cumulative sum of an m x n array, by first
+        cumulating along the y direction so that the largest values
+        are in the nth row, and then cumulating along the x direction,
+        so that the largest values are in the mth column. Each grid point
+        will contain the cumulative sum from the origin to that grid point.
+
+        Parameters
+        ----------
+        cube : Iris.cube.Cube
+            Cube to which the cumulative summing along the y and x direction
+            will be applied.
+
+        Returns
+        -------
+        cube : Iris.cube.Cube
+            Cube to which the cumulative summing along the y and x direction
+            has been applied.
+        nan_masks : list
+            List of numpy arrays to be used to set the values within the data
+            of the output cube to be NaN.
+        """
+        yname = cube.coord(axis="y").name()
+        xname = cube.coord(axis="x").name()
+        cubelist = iris.cube.CubeList([])
+        nan_masks = []
+        for slice_2d in cube.slices([yname, xname]):
+            data = slice_2d.data
+            nan_mask = np.isnan(data)
+            data[nan_mask] = 0
+            data_summed_along_y = np.cumsum(data, axis=0)
+            data_summed_along_x = (
+                np.cumsum(data_summed_along_y, axis=1))
+            slice_2d.data = data_summed_along_x
+            cubelist.append(slice_2d)
+            nan_masks.append(nan_mask)
+        return cubelist.merge_cube(), nan_masks
+
+    @staticmethod
+    def _check_for_x_and_y_axes(cube):
```

Comment:
I know this is a trivial little check, but it's exactly the kind of simple generic function that will be endlessly duplicated if we don't put it in utilities. Please move it there.